tests/main/snap-confine-undesired-mode-group/task.yaml

```diff
@@ -0,0 +1,48 @@
+summary: the snap-{run,confine,exec} chain does not create files with undesired properties.
+prepare: |
+    # Install a snap with opengl and joystick plugs.
+    # This gives us all of the usual snap-confine configuration, along with all
+    # the cgroups that we create.
+    snap pack test-snapd-app
+    snap install --dangerous ./test-snapd-app_1.0_all.snap
+    snap connect test-snapd-app:opengl
+    snap connect test-snapd-app:joystick
+execute: |
+    # Run the snap as a non-root user.
+    su test -c 'snap run test-snapd-app.sh -c /bin/true'
+
+    # Look for files that are owned by the test user, group owned by the test
+    # user or are world-writable in /run/snapd as well as in /sys/fs/cgroup
+    # trees.
+    for dname in /run/snapd /sys/fs/cgroup; do
```

Comment:
Not that it should be added right here, but I recall the issue of the dir in /tmp ending up with wrong permissions. IIRC, that is also in another spread test. I wonder if it makes sense to have it here (too?) since this spread test is meant to handle all the cases. Not a blocker if there is already a test for this.

Reply:
That's a great idea. I'll extend it to scan /tmp. if it finds anything this needs to be fixed as well